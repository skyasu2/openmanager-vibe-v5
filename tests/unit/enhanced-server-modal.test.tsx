import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import EnhancedServerModal from '@/components/dashboard/EnhancedServerModal';

const mockServer = {
  id: 'api-eu-045',
  hostname: 'api-eu-045.internal',
  name: 'api-eu-045',
  type: 'api',
  environment: 'production',
  location: 'EU West',
  provider: 'AWS',
  status: 'warning' as const,
  cpu: 67,
  memory: 72,
  disk: 54,
  network: 45,
  uptime: '8일 12시간',
  lastUpdate: new Date(),
  alerts: 2,
  services: [
    { name: 'nodejs', status: 'running' as const, port: 3000 },
    { name: 'nginx', status: 'running' as const, port: 80 },
  ],
  specs: {
    cpu_cores: 4,
    memory_gb: 16,
    disk_gb: 100,
  },
  os: 'Ubuntu 22.04',
  ip: '10.0.0.12',
  networkStatus: 'good' as const,
};

describe('EnhancedServerModal', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = render(
      <EnhancedServerModal server={mockServer} onClose={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
