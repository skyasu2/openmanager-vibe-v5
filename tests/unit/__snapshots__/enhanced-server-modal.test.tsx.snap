// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EnhancedServerModal > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="fixed inset-0 bg-black/70 backdrop-blur-sm z-50 flex items-center justify-center p-4"
    style="opacity: 0;"
  >
    <div
      class="bg-white rounded-2xl shadow-2xl border border-gray-200 w-full max-w-6xl h-[90vh] flex flex-col overflow-hidden ring-1 ring-black/5"
      style="opacity: 0; transform: translateY(20px) scale(0.9);"
    >
      <div
        class="bg-gradient-to-r from-blue-600 to-purple-600 text-white p-6"
      >
        <div
          class="flex items-center justify-between"
        >
          <div
            class="flex items-center gap-4"
          >
            <div
              class="p-3 bg-white/20 rounded-lg"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-server w-6 h-6"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="8"
                  rx="2"
                  ry="2"
                  width="20"
                  x="2"
                  y="2"
                />
                <rect
                  height="8"
                  rx="2"
                  ry="2"
                  width="20"
                  x="2"
                  y="14"
                />
                <line
                  x1="6"
                  x2="6.01"
                  y1="6"
                  y2="6"
                />
                <line
                  x1="6"
                  x2="6.01"
                  y1="18"
                  y2="18"
                />
              </svg>
            </div>
            <div>
              <h2
                class="text-2xl font-bold flex items-center gap-3"
              >
                <span>
                  api-eu-045
                </span>
              </h2>
              <p
                class="text-blue-100 flex items-center gap-2"
              >
                api
                 • 
                EU West
              </p>
            </div>
          </div>
          <div
            class="flex items-center gap-3"
          >
            <button
              class="px-4 py-2 rounded-lg flex items-center gap-2 transition-all duration-200 bg-green-500 shadow-lg"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-play w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="6 3 20 12 6 21 6 3"
                />
              </svg>
              실시간
            </button>
            <button
              class="p-2 bg-white/30 backdrop-blur-sm rounded-lg hover:bg-white/40 transition-all duration-200"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-x w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="flex gap-2 mt-6"
        >
          <button
            class="px-4 py-2 rounded-lg flex items-center gap-2 transition-all duration-200 bg-white text-blue-600 shadow-lg ring-1 ring-blue-200"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-activity w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 12h-2.48a2 2 0 0 0-1.93 1.46l-2.35 8.36a.25.25 0 0 1-.48 0L9.24 2.18a.25.25 0 0 0-.48 0l-2.35 8.36A2 2 0 0 1 4.49 12H2"
              />
            </svg>
            개요
          </button>
          <button
            class="px-4 py-2 rounded-lg flex items-center gap-2 transition-all duration-200 bg-white/20 backdrop-blur-sm text-white hover:bg-white/30"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-chart-column w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 3v16a2 2 0 0 0 2 2h16"
              />
              <path
                d="M18 17V9"
              />
              <path
                d="M13 17V5"
              />
              <path
                d="M8 17v-3"
              />
            </svg>
            메트릭
          </button>
          <button
            class="px-4 py-2 rounded-lg flex items-center gap-2 transition-all duration-200 bg-white/20 backdrop-blur-sm text-white hover:bg-white/30"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-cpu w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 20v2"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M17 20v2"
              />
              <path
                d="M17 2v2"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M2 17h2"
              />
              <path
                d="M2 7h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="M20 17h2"
              />
              <path
                d="M20 7h2"
              />
              <path
                d="M7 20v2"
              />
              <path
                d="M7 2v2"
              />
              <rect
                height="16"
                rx="2"
                width="16"
                x="4"
                y="4"
              />
              <rect
                height="8"
                rx="1"
                width="8"
                x="8"
                y="8"
              />
            </svg>
            프로세스
          </button>
          <button
            class="px-4 py-2 rounded-lg flex items-center gap-2 transition-all duration-200 bg-white/20 backdrop-blur-sm text-white hover:bg-white/30"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-file-text w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"
              />
              <path
                d="M14 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M10 9H8"
              />
              <path
                d="M16 13H8"
              />
              <path
                d="M16 17H8"
              />
            </svg>
            로그
          </button>
          <button
            class="px-4 py-2 rounded-lg flex items-center gap-2 transition-all duration-200 bg-white/20 backdrop-blur-sm text-white hover:bg-white/30"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-network w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="6"
                rx="1"
                width="6"
                x="16"
                y="16"
              />
              <rect
                height="6"
                rx="1"
                width="6"
                x="2"
                y="16"
              />
              <rect
                height="6"
                rx="1"
                width="6"
                x="9"
                y="2"
              />
              <path
                d="M5 16v-3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3"
              />
              <path
                d="M12 12V8"
              />
            </svg>
            네트워크
          </button>
        </div>
      </div>
      <div
        class="flex-1 p-6 overflow-y-auto bg-gradient-to-br from-gray-50 to-gray-100"
      >
        <div
          style="opacity: 0; transform: translateX(20px);"
        >
          <div
            class="space-y-6"
          >
            <div>
              <h3
                class="text-xl font-bold text-gray-900 mb-4"
              >
                실시간 리소스 모니터링
              </h3>
              <div
                class="grid grid-cols-1 md:grid-cols-3 gap-8 bg-white rounded-xl p-6 shadow-md border border-gray-200"
              >
                <div
                  class="flex flex-col items-center"
                >
                  <div
                    class="relative"
                    style="width: 140px; height: 140px;"
                  >
                    <svg
                      class="transform -rotate-90 drop-shadow-lg"
                      height="140"
                      width="140"
                    >
                      <circle
                        class="drop-shadow-sm"
                        cx="70"
                        cy="70"
                        fill="transparent"
                        r="45"
                        stroke="#e5e7eb"
                        stroke-width="8"
                      />
                      <circle
                        class="transition-all duration-1000 ease-out drop-shadow-sm"
                        cx="70"
                        cy="70"
                        fill="transparent"
                        r="45"
                        stroke="#ef4444"
                        stroke-dasharray="282.7433388230814"
                        stroke-dashoffset="93.30530181161686"
                        stroke-linecap="round"
                        stroke-width="8"
                        style="filter: drop-shadow(0 0 8px #ef444440);"
                      />
                      <circle
                        class="opacity-20"
                        cx="70"
                        cy="70"
                        fill="url(#gradient-CPU)"
                        r="35"
                      />
                      <defs>
                        <radialgradient
                          id="gradient-CPU"
                        >
                          <stop
                            offset="0%"
                            stop-color="#ef4444"
                            stop-opacity="0.3"
                          />
                          <stop
                            offset="100%"
                            stop-color="#ef4444"
                            stop-opacity="0.1"
                          />
                        </radialgradient>
                      </defs>
                    </svg>
                    <div
                      class="absolute inset-0 flex items-center justify-center"
                    >
                      <div
                        class="text-center"
                      >
                        <div
                          class="text-2xl font-bold"
                          style="color: rgb(239, 68, 68);"
                        >
                          67
                          %
                        </div>
                        <div
                          class="text-xs text-gray-500 mt-1"
                        >
                          CPU
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="flex flex-col items-center"
                >
                  <div
                    class="relative"
                    style="width: 140px; height: 140px;"
                  >
                    <svg
                      class="transform -rotate-90 drop-shadow-lg"
                      height="140"
                      width="140"
                    >
                      <circle
                        class="drop-shadow-sm"
                        cx="70"
                        cy="70"
                        fill="transparent"
                        r="45"
                        stroke="#e5e7eb"
                        stroke-width="8"
                      />
                      <circle
                        class="transition-all duration-1000 ease-out drop-shadow-sm"
                        cx="70"
                        cy="70"
                        fill="transparent"
                        r="45"
                        stroke="#3b82f6"
                        stroke-dasharray="282.7433388230814"
                        stroke-dashoffset="79.16813487046281"
                        stroke-linecap="round"
                        stroke-width="8"
                        style="filter: drop-shadow(0 0 8px #3b82f640);"
                      />
                      <circle
                        class="opacity-20"
                        cx="70"
                        cy="70"
                        fill="url(#gradient-메모리)"
                        r="35"
                      />
                      <defs>
                        <radialgradient
                          id="gradient-메모리"
                        >
                          <stop
                            offset="0%"
                            stop-color="#3b82f6"
                            stop-opacity="0.3"
                          />
                          <stop
                            offset="100%"
                            stop-color="#3b82f6"
                            stop-opacity="0.1"
                          />
                        </radialgradient>
                      </defs>
                    </svg>
                    <div
                      class="absolute inset-0 flex items-center justify-center"
                    >
                      <div
                        class="text-center"
                      >
                        <div
                          class="text-2xl font-bold"
                          style="color: rgb(59, 130, 246);"
                        >
                          72
                          %
                        </div>
                        <div
                          class="text-xs text-gray-500 mt-1"
                        >
                          메모리
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="flex flex-col items-center"
                >
                  <div
                    class="relative"
                    style="width: 140px; height: 140px;"
                  >
                    <svg
                      class="transform -rotate-90 drop-shadow-lg"
                      height="140"
                      width="140"
                    >
                      <circle
                        class="drop-shadow-sm"
                        cx="70"
                        cy="70"
                        fill="transparent"
                        r="45"
                        stroke="#e5e7eb"
                        stroke-width="8"
                      />
                      <circle
                        class="transition-all duration-1000 ease-out drop-shadow-sm"
                        cx="70"
                        cy="70"
                        fill="transparent"
                        r="45"
                        stroke="#8b5cf6"
                        stroke-dasharray="282.7433388230814"
                        stroke-dashoffset="130.06193585861743"
                        stroke-linecap="round"
                        stroke-width="8"
                        style="filter: drop-shadow(0 0 8px #8b5cf640);"
                      />
                      <circle
                        class="opacity-20"
                        cx="70"
                        cy="70"
                        fill="url(#gradient-디스크)"
                        r="35"
                      />
                      <defs>
                        <radialgradient
                          id="gradient-디스크"
                        >
                          <stop
                            offset="0%"
                            stop-color="#8b5cf6"
                            stop-opacity="0.3"
                          />
                          <stop
                            offset="100%"
                            stop-color="#8b5cf6"
                            stop-opacity="0.1"
                          />
                        </radialgradient>
                      </defs>
                    </svg>
                    <div
                      class="absolute inset-0 flex items-center justify-center"
                    >
                      <div
                        class="text-center"
                      >
                        <div
                          class="text-2xl font-bold"
                          style="color: rgb(139, 92, 246);"
                        >
                          54
                          %
                        </div>
                        <div
                          class="text-xs text-gray-500 mt-1"
                        >
                          디스크
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="grid grid-cols-1 md:grid-cols-2 gap-6"
            >
              <div
                class="bg-white rounded-xl p-6 shadow-md border border-gray-200"
              >
                <h4
                  class="text-lg font-semibold text-gray-900 mb-4"
                >
                  시스템 정보
                </h4>
                <div
                  class="space-y-3"
                >
                  <div
                    class="flex justify-between"
                  >
                    <span
                      class="text-gray-600"
                    >
                      운영체제
                    </span>
                    <span
                      class="font-medium"
                    >
                      Ubuntu 22.04
                    </span>
                  </div>
                  <div
                    class="flex justify-between"
                  >
                    <span
                      class="text-gray-600"
                    >
                      IP 주소
                    </span>
                    <span
                      class="font-medium"
                    >
                      10.0.0.12
                    </span>
                  </div>
                  <div
                    class="flex justify-between"
                  >
                    <span
                      class="text-gray-600"
                    >
                      업타임
                    </span>
                    <span
                      class="font-medium"
                    >
                      8일 12시간
                    </span>
                  </div>
                  <div
                    class="flex justify-between"
                  >
                    <span
                      class="text-gray-600"
                    >
                      CPU 코어
                    </span>
                    <span
                      class="font-medium"
                    >
                      4
                      개
                    </span>
                  </div>
                  <div
                    class="flex justify-between"
                  >
                    <span
                      class="text-gray-600"
                    >
                      메모리
                    </span>
                    <span
                      class="font-medium"
                    >
                      16
                      GB
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="bg-white rounded-xl p-6 shadow-md border border-gray-200"
              >
                <h4
                  class="text-lg font-semibold text-gray-900 mb-4"
                >
                  서비스 상태
                </h4>
                <div
                  class="space-y-3"
                >
                  <div
                    class="flex items-center justify-between"
                  >
                    <div
                      class="flex items-center gap-3"
                    >
                      <div
                        class="w-3 h-3 rounded-full bg-green-500"
                      />
                      <span
                        class="font-medium"
                      >
                        nodejs
                      </span>
                    </div>
                    <span
                      class="text-sm text-gray-600"
                    >
                      :
                      3000
                    </span>
                  </div>
                  <div
                    class="flex items-center justify-between"
                  >
                    <div
                      class="flex items-center gap-3"
                    >
                      <div
                        class="w-3 h-3 rounded-full bg-green-500"
                      />
                      <span
                        class="font-medium"
                      >
                        nginx
                      </span>
                    </div>
                    <span
                      class="text-sm text-gray-600"
                    >
                      :
                      80
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
